cfgov/ask_cfpb/views.py

```diff
@@ -143,19 +143,22 @@ def ask_search_es7(request, language='en', as_json=False):
 
     # If there's no query string, don't search
     search_term = request.GET.get('q', '')
-    # Check if we want to use the suggestion or not
-    suggest = request.GET.get('correct', '1') == '1'
     if not search_term:
         results_page.query = ''
         results_page.result_query = ''
         return results_page.serve(request)
 
     page = AnswerPageSearch(search_term, language=language)
-    if suggest:
-        suggestion = page.suggest().get('suggestion')
-
     response = page.search()
-    if response.get('results'):
+
+    # Check if we want to use the suggestion or not
+    suggest = request.GET.get('correct', '1') == '1'
+
+    # Provide a suggestion only when no results are found
+    if not response.get('results') and suggest:
+        response = page.suggest()
+        suggestion = response.get('suggestion')
```

Comment:
I don't think you're getting coverage on these lines because `mock.patch.object` returns a `MagicMock`, and you're not providing a `get` mock on the `mock_suggestion` return value... So, `response.get('results')` returns a truthy value and this code doesn't get called in the test.

You might want to check what `response.get('results')` returns when running the `test_en_search_suggestion_es7` test to make sure of that, but that's my guess.

Reply:
I checked that `response.get('results')` is `None` when running `test_en_search_suggestion_es7`.

I got errors when I tried to add the following line of code so I commented it out:
`mock_search.return_value = [mock_es_queryset]`

I also tried the following line of code but that had errors as well:
`mock_search.return_value = [mock_return]`

Reply:
Updated suggestion test to get coverage on lines 159 and 160 in views.py